--- a/ap.c
+++ b/ap.c
@@ -4828,28 +4828,13 @@ static int cmd_ap_config_commit(struct s
 	case AP_11g:
 	case AP_11b:
 	case AP_11ng:
-		ifname = (drv == DRIVER_MAC80211 || drv == DRIVER_LINUX_WCN) ?
-			"wlan0" : "ath0";
-		if (drv == DRIVER_QNXNTO && sigma_main_ifname)
-			ifname = sigma_main_ifname;
+		ifname = get_main_ifname();
 		fprintf(f, "hw_mode=g\n");
 		break;
 	case AP_11a:
 	case AP_11na:
 	case AP_11ac:
-		if (drv == DRIVER_QNXNTO) {
-			if (sigma_main_ifname)
-				ifname = sigma_main_ifname;
-			else
-				ifname = "wlan0";
-		} else if (drv == DRIVER_MAC80211 || drv == DRIVER_LINUX_WCN) {
-			if (if_nametoindex("wlan1") > 0)
-				ifname = "wlan1";
-			else
-				ifname = "wlan0";
-		} else {
-			ifname = get_main_ifname();
-		}
+		ifname = get_main_ifname();
 		fprintf(f, "hw_mode=a\n");
 		break;
 	default:
@@ -5846,9 +5831,9 @@ static int cmd_ap_get_info(struct sigma_
 		else
 			version = "Unknown";
 
-		if (if_nametoindex("wlan1") > 0)
-			snprintf(resp, sizeof(resp), "interface,wlan0_24G "
-				 "wlan1_5G,agent,1.0,version,%s", version);
+		if (if_nametoindex("wlan0") > 0)
+			snprintf(resp, sizeof(resp), "interface,wlan0_5G "
+				 "wlan1_24G,agent,1.0,version,%s", version);
 		else
 			snprintf(resp, sizeof(resp), "interface,wlan0_any,"
 				 "agent,1.0,version,%s", version);
@@ -5962,12 +5947,7 @@ static int ap_inject_frame(struct sigma_
 	char cbuf[100];
 	struct ifreq ifr;
 
-	if ((dut->ap_mode == AP_11a || dut->ap_mode == AP_11na ||
-	     dut->ap_mode == AP_11ac) &&
-	    if_nametoindex("wlan1") > 0)
-		ifname = "wlan1";
-	else
-		ifname = "wlan0";
+	ifname = get_main_ifname();
 
 	if (hwaddr_aton(sta_addr, addr_sta) < 0)
 		return -1;
@@ -6567,12 +6547,7 @@ static int cmd_ap_get_mac_address(struct
 		else
 			ifname = "ath0";
 	} else {
-		if ((dut->ap_mode == AP_11a || dut->ap_mode == AP_11na ||
-		     dut->ap_mode == AP_11ac) &&
-		    if_nametoindex("wlan1") > 0)
-			ifname = "wlan1";
-		else
-			ifname = "wlan0";
+		ifname = get_main_ifname();
 	}
 
 	s = socket(AF_INET, SOCK_DGRAM, 0);
